ipld/read_test.go

```diff
@@ -231,3 +234,66 @@ func removeRandShares(data [][]byte, d int) [][]byte {
 	}
 	return data
 }
+
+func TestGetSharesByNamespace(t *testing.T) {
+	var tests = []struct {
+		rawData [][]byte
+	}{
+		{rawData: testutils.GenerateRandNamespacedRawData(16, NamespaceSize, plugin.ShareSize)},
+		{rawData: testutils.GenerateRandNamespacedRawData(16, NamespaceSize, 8)},
+		{rawData: testutils.GenerateRandNamespacedRawData(4, NamespaceSize, plugin.ShareSize)},
+		{rawData: testutils.GenerateRandNamespacedRawData(16, NamespaceSize, 8)},
+	}
+
+	for i, tt := range tests {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			squareSize := uint64(math.Sqrt(float64(len(tt.rawData))))
+
+			// choose random nID from rand shares
+			expected := tt.rawData[len(tt.rawData)/2]
+			nID := expected[:8]
+
+			// change rawData to contain several shares with same nID
+			tt.rawData[(len(tt.rawData)/2)+1] = expected
+
+			// generate DAH
+			dah, err := da.NewDataAvailabilityHeader(squareSize, tt.rawData)
+			require.NoError(t, err)
+
+			// put raw data in DAG
+			dag := mdutils.Mock()
+			_, err = PutData(context.Background(), tt.rawData, dag)
+			require.NoError(t, err)
+
+			rowRootCIDs, err := rowRootsByNamespaceID(nID, &dah)
+			require.NoError(t, err)
+
+			for _, rowCID := range rowRootCIDs {
+				nodes, err := GetLeavesByNamespace(context.Background(), dag, rowCID, nID)
+				require.NoError(t, err)
```

Comment:
Similar to below: Need test cases for the not-found-cases here as well. Can also happen in a follow-up PR if there is an issue for that.